### PR TITLE
added material and news JOI validation

### PR DESCRIPTION
**Pull request by ostgit12**

### CODE REVIEW

File: modules/news/news.permissions.js [hunk 1]
```diff
@@ -1,17 +1,30 @@
-const { allow } = require('graphql-shield');
+const { allow, and } = require('graphql-shield');
 const { hasRoles } = require('../../utils/rules');
 const { roles } = require('../../consts');
+const { newsInputValidator } = require('../../validators/news.validator');
+const { inputDataValidation } = require('../../utils/rules');
+
 const { ADMIN, SUPERADMIN } = roles;
```
Comment (yuliiakucher) [inline modules/news/news.permissions.js:7]: same

File: validators/material.validator.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+const {
+  materialPurposes: { MAIN, INNER, BOTTOM, PATTERN, CLOSURE },
+} = require('../consts/material-purposes');
+
+const languageSchemaValidator = {
+  lang: Joi.string()
+    .trim()
+    .required(),
+  value: Joi.string().trim(),
+};
```
Comment (yuliiakucher) [inline validators/material.validator.js:12]: please don't duplicate this part, create languageValidator for this case

File: validators/material.validator.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+const {
+  materialPurposes: { MAIN, INNER, BOTTOM, PATTERN, CLOSURE },
+} = require('../consts/material-purposes');
+
+const languageSchemaValidator = {
+  lang: Joi.string()
+    .trim()
+    .required(),
+  value: Joi.string().trim(),
```
Comment (yuliiakucher) [inline validators/material.validator.js:11]: please add min, max and required

File: validators/material.validator.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const Joi = require('joi');
+
+const {
+  materialPurposes: { MAIN, INNER, BOTTOM, PATTERN, CLOSURE },
+} = require('../consts/material-purposes');
+
+const languageSchemaValidator = {
```
Comment (yuliiakucher) [inline validators/material.validator.js:7]: please change languageSchemaValidator into a function, that will take min and max as arguments. It should return joi schema.

File: validators/news.validator.js [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  value: Joi.string().trim(),
+};
+
+const newsInputValidator = Joi.object({
+  title: Joi.array().items(languageSchemaValidator),
+  text: Joi.array().items(languageSchemaValidator),
+  image: Joi.string().trim(),
+  author: Joi.object({
+    name: Joi.array().items(languageSchemaValidator),
+    image: Joi.string().trim(),
```
Comment (yuliiakucher) [inline validators/news.validator.js:16]: please don't validate image

**Review by yuliiakucher — CHANGES_REQUESTED** [review]
please fix comments

**Comment by RevylaV** [comment]
Fix tests please

**Review by Elugormo — DISMISSED** [review]
Resolve conflicts on your branch please and update it to the current development origin version. And check the CI.
Please write a proper description and please unbind PR from the project.

File: modules/material/material.permissions.js [hunk 6]
```diff
@@ -1,17 +1,32 @@
+const { allow, and } = require('graphql-shield');
 const { hasRoles } = require('../../utils/rules');
 const { roles } = require('../../consts');
+const {
+  materialInputValidator,
+} = require('../../validators/material.validator');
+const { inputDataValidation } = require('../../utils/rules');
 
 const { ADMIN, SUPERADMIN } = roles;
```
[the remark this answers is not in the document]
Comment (ostgit12) [reply modules/material/material.permissions.js]: destruct row 3

**Review by yuliiakucher — APPROVED** [review]

**Review by Skrekliam — APPROVED** [review]

**Review by martagorpinchuk — APPROVED** [review]

**Pull request merged**